src/Core/Util/Entity/EntityNameDuplicator.php

```diff
@@ -0,0 +1,100 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Util\Entity;
+
+use Language;
+use PrestaShop\PrestaShop\Core\Util\String\StringModifierInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class EntityNameDuplicator
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var StringModifierInterface
+     */
+    private $stringModifier;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param StringModifierInterface $stringModifier
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        StringModifierInterface $stringModifier
+    ) {
+        $this->translator = $translator;
+        $this->stringModifier = $stringModifier;
+    }
+
+    /**
+     * Adds a "copy" word to localized entity names
+     *
+     * @param array<int, string> $currentLocalizedNames Array with localized names, IdLang => Name
+     * @param int $maxLength Max length of a new name, will be cut if it doesn't fit with the new "copy" word.
+     *
+     * @return array<int, string>
+     */
+    public function getNewLocalizedNames(array $currentLocalizedNames, int $maxLength): array
+    {
+        $newLocalizedNames = [];
+        foreach ($currentLocalizedNames as $langId => $currentName) {
+            $langId = (int) $langId;
+
+            // Add copy word to the string
+            $namePattern = $this->translator->trans('copy of %s', [], 'Admin.Catalog.Feature', Language::getLocaleById($langId));
+            $newName = sprintf($namePattern, $currentName);
+            
+            // Save and cut the end if it doesn't fit to the limit
+            $newLocalizedNames[$langId] = $this->stringModifier->cutEnd($newName, $maxLength);
+        }
+
+        return $newLocalizedNames;
+    }
+
+    /**
+     * Appends a "copy" to simple entity name
+     *
+     * @param string $currentName Current name of the entity
+     * @param int $maxLength Max length of a new name, will be cut if it doesn't fit with the new "copy" word.
+     * @param int $idLang
+     *
+     * @return string
+     */
+    public function getNewName(string $currentName, int $maxLength, int $idLang): string
+    {
+        // Add copy word to the string
+        $namePattern = $this->translator->trans('copy of %s', [], 'Admin.Catalog.Feature', Language::getLocaleById($idLang));
```

Comment:
following above suggestion
```suggestion
        $namePattern = $this->translator->trans('copy of %s', [], 'Admin.Catalog.Feature', Language::getLocaleById($langId));
```